### PR TITLE
Add crd: true prop to pipeline models

### PR DESCRIPTION
**Pull request by vikram-raj**
Fix bug - https://jira.coreos.com/browse/ODC-1916

### CODE REVIEW

**Comment by spadgett** [comment]
You probably want `referenceForModel(PipelineModel)` here: 

https://github.com/openshift/console/blob/master/frontend/packages/dev-console/src/plugin.tsx#L93

**Comment by vikram-raj** [comment]
> You probably want `referenceForModel(PipelineModel)` here:
> 
> https://github.com/openshift/console/blob/master/frontend/packages/dev-console/src/plugin.tsx#L93

@spadgett @christianvogt Event after adding ```resource: referenceForModel(PipelineModel)```. I am getting these warning 

```
kindObj: no model for kind Pipeline
No model registered for Pipeline
```

Am I missing something?

**Comment by vikram-raj** [comment]
@spadgett Ready for review again.

**Comment by spadgett** [comment]
We also need to fix:

https://github.com/vikram-raj/console/blob/d8611043a5bf691e209cfc63062dd453af6f2c29/frontend/packages/dev-console/src/utils/pipeline-actions.tsx#L31

We should be using `resourceListPathFromModel` instead of building the URL ourselves:

https://github.com/vikram-raj/console/blob/d8611043a5bf691e209cfc63062dd453af6f2c29/frontend/public/components/utils/resource-link.tsx#L40

File: frontend/packages/dev-console/src/models/pipelines.ts [hunk 1]
```diff
@@ -10,6 +10,7 @@ export const PipelineModel: K8sKind = {
   kind: 'Pipeline',
   id: 'pipeline',
   labelPlural: 'Pipelines',
+  crd: true,
```
Comment (spadgett) [inline frontend/packages/dev-console/src/models/pipelines.ts]: Shouldn't the other pipeline models have `crd: true`?

Comment (vikram-raj) [reply frontend/packages/dev-console/src/models/pipelines.ts]: Added ```crd: true``` to all models.

**Comment by vikram-raj** [comment]
@spadgett Ready for review again.

File: frontend/packages/dev-console/src/components/pipelineruns/PipelineRunVisualization.tsx [hunk 2]
```diff
@@ -21,7 +21,7 @@ export class PipelineRunVisualization extends React.Component<
   constructor(props) {
     super(props);
     this.state = {
-      pipeline: { apiVersion: '', metadata: {}, kind: 'PipelineRun' },
+      pipeline: { apiVersion: '', metadata: {}, kind: referenceForModel(PipelineResourceModel) },
```
Comment (spadgett) [inline frontend/packages/dev-console/src/components/pipelineruns/PipelineRunVisualization.tsx]: You don't want `referenceForModel` here. This should be just the kind as before

File: frontend/packages/dev-console/src/utils/pipeline-actions.tsx [hunk 3]
```diff
@@ -39,7 +40,7 @@ export const getPipelineRunData = (pipeline: Pipeline, latestRun?: PipelineRun):
   // TODO: Add serviceAccount for start scenario by fetching details from user
   return {
     apiVersion: `${PipelineRunModel.apiGroup}/${PipelineRunModel.apiVersion}`,
-    kind: PipelineRunModel.kind,
+    kind: referenceForModel(PipelineRunModel),
```
Comment (spadgett) [inline frontend/packages/dev-console/src/utils/pipeline-actions.tsx]: This was correct before

File: frontend/packages/dev-console/src/utils/pipeline-augment.ts [hunk 4]
```diff
@@ -110,7 +111,7 @@ export const getResources = (data: PropPipelineData[]): Resource => {
       if (pipeline.metadata && pipeline.metadata.namespace && pipeline.metadata.name) {
         propsReferenceForRuns.push(`PipelineRun_${i}`);
         resources.push({
-          kind: 'PipelineRun',
+          kind: referenceForModel(PipelineRunModel),
```
Comment (spadgett) [inline frontend/packages/dev-console/src/utils/pipeline-augment.ts]: This was correct before

File: frontend/packages/dev-console/src/utils/pipeline-utils.ts [hunk 5]
```diff
@@ -137,7 +138,11 @@ const appendPipelineRunStatus = (pipeline, pipelineRun) => {
 
 export const getPipelineTasks = (
   pipeline: K8sResourceKind,
-  pipelineRun: K8sResourceKind = { apiVersion: '', metadata: {}, kind: 'PipelineRun' },
+  pipelineRun: K8sResourceKind = {
+    apiVersion: '',
+    metadata: {},
+    kind: referenceForModel(PipelineResourceModel),
```
Comment (spadgett) [inline frontend/packages/dev-console/src/utils/pipeline-utils.ts]: `kind` should still be `PipelineRun`

**Comment by spadgett** [comment]
We might want to push to 4.3 or leave well enough alone. I'm a little worried we'll introduce bugs here.

File: frontend/packages/dev-console/src/utils/pipeline-augment.ts (hunk 4 above)
Comment (vikram-raj) [reply frontend/packages/dev-console/src/utils/pipeline-augment.ts]: Getting error ```No model registered for PipelineRun```. If I just pass ```'PipelineRun'````

File: frontend/packages/dev-console/src/components/pipelineruns/PipelineRunVisualization.tsx (hunk 2 above)
Comment (vikram-raj) [reply frontend/packages/dev-console/src/components/pipelineruns/PipelineRunVisualization.tsx]: Done.

File: frontend/packages/dev-console/src/utils/pipeline-actions.tsx (hunk 3 above)
Comment (vikram-raj) [reply frontend/packages/dev-console/src/utils/pipeline-actions.tsx]: Done

File: frontend/packages/dev-console/src/utils/pipeline-utils.ts (hunk 5 above)
Comment (vikram-raj) [reply frontend/packages/dev-console/src/utils/pipeline-utils.ts]: Done

File: frontend/packages/dev-console/src/utils/pipeline-augment.ts (hunk 4 above)
Comment (spadgett) [reply frontend/packages/dev-console/src/utils/pipeline-augment.ts]: Oops, yeah this is for `Firehose`. Your change is right here. My fault.

We should rename that prop honestly. It's not a `kind`.

**Comment by christianvogt** [comment]
> We might want to push to 4.3 or leave well enough alone. I'm a little worried we'll introduce bugs here.

I agree. This turned out to be a much larger change than anticipated.

File: frontend/packages/dev-console/src/components/pipelines/pipeline-resource/pipelineResource-utils.ts [hunk 6]
```diff
@@ -27,7 +27,7 @@ export const createPipelineResource = (
   const resourceName = `${type}-${getRandomChars(6)}`;
   const pipelineResource: K8sResourceKind = {
     apiVersion: 'tekton.dev/v1alpha1',
-    kind: 'PipelineResource',
+    kind: referenceForModel(PipelineResourceModel),
```
Comment (christianvogt) [inline frontend/packages/dev-console/src/components/pipelines/pipeline-resource/pipelineResource-utils.ts]: @vikram-raj This should be left alone or changed to:
```suggestion
    kind: PipelineResourceModel.kind,
```

It's odd though. I tested the code and although it creates a `kind` with value `tekton.dev~v1alpha1~PipelineResource`, the POST request to create the resource works.

**Comment by christianvogt** [comment]
Tested many scenarios for pipelines and everything worked. Found change that should be corrected.

**Comment by spadgett** [comment]
Retargeting to master-4.3, but you'll need to rebase or wait for #2803

**Comment by vikram-raj** [comment]
@spadgett @christianvogt As the master is now updated with the master-4.3. Can we move forward with this PR now?

**Pull request merged**